test_decorators.py

```diff
@@ -0,0 +1,242 @@
+        """
+        Method decorators are accumulated.
+        """
+        code = """
+class Foo:
+    @first_decorator
+    def method(self):
+        pass
+        """
+
+        module = pep257.parse(StringIO(code), 'dummy.py')
+        defined_class = module.children[0]
+        decorators = defined_class.children[0].decorators
+
+        assert 1 == len(decorators)
+        assert 'first_decorator' == decorators[0].name
+        assert '' == decorators[0].arguments
+
+    def test_parse_method_decorators(self):
+        """
+        Method decorators are accumulated.
+        """
+        code = """
+class Foo:
+    @first_decorator
+    @second.decorator(argument)
+    @third.multi.line(
+        decorator,
+        key=value,
+        )
+    def method(self):
+        pass
+        """
+
+        module = pep257.parse(StringIO(code), 'dummy.py')
+        defined_class = module.children[0]
+        decorators = defined_class.children[0].decorators
+
+        assert 3 == len(decorators)
+        assert 'first_decorator' == decorators[0].name
+        assert '' == decorators[0].arguments
+        assert 'second.decorator' == decorators[1].name
+        assert 'argument' == decorators[1].arguments
+        assert 'third.multi.line' == decorators[2].name
+        assert 'decorator,key=value,' == decorators[2].arguments
+
+    def test_parse_function_decorator(self):
+        """
+        It accumulates decorators for functions.
+        """
+        code = """@first_decorator
+def some_method(self):
+    pass
+        """
+
+        module = pep257.parse(StringIO(code), 'dummy.py')
+        decorators = module.children[0].decorators
+
+        assert 1 == len(decorators)
+        assert 'first_decorator' == decorators[0].name
+        assert '' == decorators[0].arguments
+
+    def test_parse_method_nested_decorator(self):
+        """
+        Method decorators are accumulated for nested methods.
+        """
+        code = """
+class Foo:
+    @parent_decorator
+    def method(self):
+        @first_decorator
+        def nested_method(arg):
+            pass
+        """
+
+        module = pep257.parse(StringIO(code), 'dummy.py')
+        defined_class = module.children[0]
+        decorators = defined_class.children[0].children[0].decorators
+
+        assert 1 == len(decorators)
+        assert 'first_decorator' == decorators[0].name
+        assert '' == decorators[0].arguments
+
+
+class TestMethod:
+    """
+    Unit test for Method class.
+    """
+
+    def makeMethod(self, name='someMethodName'):
+        """
+        Return a simple method instance.
+        """
+        children = []
+        all = ['ClassName']
+        source = 'class ClassName:\n def %s(self):\n' % (name)
+
+        module = pep257.Module(
+            'module_name',
+            source,
+            0,
+            1,
+            [],
+            'Docstring for module',
+            [],
+            None,
+            all,
+        )
```

Comment:
This formatting is a bit weird... If you're putting one argument per line, at least use keyword arguments.